Create API_design.md

KaoWeiCheng opened the pull request:
Please advise and guide me.
resolve https://github.com/No-57/feature/issues/3

KaoWeiCheng committed:
Create API_design.md

Please advise and guide me.

shadow3x3x3 commented on API_design.md:
```diff
@@ -0,0 +1,256 @@
+Content-Type: application/json
+Accept: application/json
+User-Agent: OrangeApp/1.0
+
```
useless empty line

shadow3x3x3 commented on API_design.md:
```diff
@@ -0,0 +1,256 @@
+Authorization: Bearer ACCESS_TOKEN
+Content-Type: application/json
+Accept: application/json
+User-Agent: OrangeApp/1.0
```
Most of time, we don't specific the `User-Agent`.

KaoWeiCheng replied:
Oh oh i got it

shadow3x3x3 commented on API_design.md:
````diff
@@ -0,0 +1,256 @@
+
+```
+### guest_header
+`Allow anonymous access without requiring authentication headers`
````
Do not use `` for normal sentense.

shadow3x3x3 commented on API_design.md:
```diff
@@ -0,0 +1,256 @@
+### URI
+`https://api.orange.com/v1/products`
+### Example 1（Get all data）：
+Request ：`Get  https://api.orange.com/v1/products`
```
```suggestion
Request: `GET https://api.orange.com/v1/products`
```

shadow3x3x3 commented on API_design.md:
````diff
@@ -0,0 +1,256 @@
+### Example 1（Get all data）：
+Request ：`Get  https://api.orange.com/v1/products`
+Response：
+```json
````
Add an empty line between sentense and block.

shadow3x3x3 commented on API_design.md:
````diff
@@ -0,0 +1,256 @@
+```json
+{
+    "code":400,
+    "errors": {
````
If using error**s** key, the value of the key should be an Array.

shadow3x3x3 commented on API_design.md:
```diff
@@ -0,0 +1,256 @@
+            "price": "30000",
+            "description": "the newst M2 Macbook 8G/256G",
+            "like": "N",
+            "image_url": "XXXXX",
+            "createAt": "2023-10-07 12:00:00:000",
```
Different naming style, please be consistency.

shadow3x3x3 commented on API_design.md:
```diff
@@ -0,0 +1,256 @@
+## Grid Table API Examples
+### URI
+`https://api.orange.com/v1/products`
+### Example 1（Get all data）：
```
I think `Example` is redundant, just use "Get all data" instead.

Even better, using "Get Products".

shadow3x3x3 commented on API_design.md:
````diff
@@ -0,0 +1,256 @@
+
+```
+
+### Methods：GET，POST
````
Don't pollute with Chinese comma "，".

shadow3x3x3 commented on API_design.md:
```diff
@@ -0,0 +1,256 @@
+ # Restful API
```
A wired space here.

```suggestion
# Restful API
```

shadow3x3x3 commented on API_design.md:
````diff
@@ -0,0 +1,256 @@
+```
+
+### Methods：GET，POST
+`https://api.orange.com/v1/products`
````
The protocol and domain are not required for API Doc. The specific API URL is enough.

```suggestion
`/v1/products`
```

shadow3x3x3 commented:
ping @Vince-Ku 

Do you have any comments on these API design? In particular, on iOS Dev.

Vince-Ku commented:
> ping @Vince-Ku
> 
> Do you have any comments on these API design? In particular, on iOS Dev.

Currently, I only need two API formats about `Products` on the `Home Page`.
- GET /products ( pagination supported is required )
- GET /products?name=xx ( fuzzy search is required )

**Optional**

- GET /{user_id}/products?field={like, xx}
( we can still discuss where the favorite products originate from, Client or Server )

Except for those formats, others can wait.

Vince-Ku commented:
Additionally, I wanna mention the concept of `Brand` and `Type`, both are also important in this project, cause I need to show all of them on the Home page.

ping @asdfg429, maybe we can discuss how to store `Brand` and `Type` in the database.

KaoWeiCheng committed:
Update API_design.md

Co-authored-by: Jeff <shadow3x3x3@gmail.com>

KaoWeiCheng committed:
Update API_design.md

The protocol and domain are not required for API Doc. The specific API URL is enough.

Co-authored-by: Jeff <shadow3x3x3@gmail.com>

KaoWeiCheng committed:
Update API_design.md

Co-authored-by: Jeff <shadow3x3x3@gmail.com>

KaoWeiCheng committed:
Update API_design.md

use shadow3x3x3 suggestion
thanks

KaoWeiCheng committed:
Update API_design.md

fix a miss sentence
